### PR TITLE
[EraVM] Rematerialize and combine 16-bit immediates

### PR DESCRIPTION
**Pull request by vladimirradosavljevic**

### CODE REVIEW

File: llvm/test/CodeGen/EraVM/fold-move-imm.ll [hunk 1]
```diff
@@ -0,0 +1,193 @@
+; NOTE: Assertions have been autogenerated by utils/update_llc_test_checks.py
+; RUN: llc -O3 < %s | FileCheck %s
+
+target datalayout = "E-p:256:256-i256:256:256-S32-a:256:256"
+target triple = "eravm"
+
+define i256 @test_pos(i256 %arg) {
```
Comment (akiramenai) [inline llvm/test/CodeGen/EraVM/fold-move-imm.ll:7]: Can we minimize the test cases?

File: llvm/lib/Target/EraVM/EraVMCombineAddressingMode.cpp [hunk 2]
```diff
@@ -553,30 +566,45 @@ bool EraVMCombineAddressingMode::combineConstantUse(MachineFunction &MF) {
   }
 
   // 2. Combine.
-  for (auto [LoadConst, Uses] : Deleted) {
+  for (auto [MoveImm, Uses] : Deleted) {
+    SmallVector<MachineOperand, 2> In0Ops;
+    // TODO: CPR-1669 If all users are COPY, don't create a constant pool entry.
```
Comment (akiramenai) [inline llvm/lib/Target/EraVM/EraVMCombineAddressingMode.cpp]: We run the pass in post-ra.

**Review by akiramenai — CHANGES_REQUESTED** [review]

File: llvm/lib/Target/EraVM/EraVMCombineAddressingMode.cpp (hunk 2 above)
Comment (vladimirradosavljevic) [reply llvm/lib/Target/EraVM/EraVMCombineAddressingMode.cpp]: Updated comment.

File: llvm/test/CodeGen/EraVM/fold-move-imm.ll (hunk 1 above)
Comment (vladimirradosavljevic) [reply llvm/test/CodeGen/EraVM/fold-move-imm.ll:7]: I saw this pattern in one of our benchmarks, and I minimize it as much as possible. I know that for this purpose is more complex than it should be, so if you want I can create smaller MIR test.

**Review by akiramenai — APPROVED** [review]

**Pull request merged**